utils/dataset_validator.py

```diff
@@ -48,72 +48,190 @@ def validate(dataset):
-                                  (DATASET_NAME_LEN_MIN, DATASET_NAME_LEN_MAX))
+    _validate_dataset_name(dataset["name"])
 
     # Description (optional)
     if "description" in dataset and dataset["description"] is not None:
-        if not isinstance(dataset["description"], string_types):
-            raise ValidationException("Value of `description` in a dataset must be a string.")
+        _validate_dataset_description(dataset["description"])
 
     # Classes
     _validate_classes(dataset["classes"])
 
     # Publicity
-    if not isinstance(dataset["public"], bool):
+    _validate_dataset_public(dataset["public"])
+
+
+def _validate_dataset_name(name):
+    """ Validate the name field of a dataset """
+    if not isinstance(name, string_types):
+        raise ValidationException("Field `name` must be a string.")
+    if not (DATASET_NAME_LEN_MIN < len(name) < DATASET_NAME_LEN_MAX):
+        raise ValidationException("Class name must be between %s and %s characters" %
+                                  (DATASET_NAME_LEN_MIN, DATASET_NAME_LEN_MAX))
+
+
+def _validate_dataset_description(description):
+    """ Validate the description of a dataset """
+    if not isinstance(description, string_types):
+        raise ValidationException("Value of `description` in a dataset must be a string.")
+
+
+def _validate_dataset_public(public):
+    """ Validate the public field of a dataset """
+    if not isinstance(public, bool):
         raise ValidationException('Value of `public` must be a boolean.')
 
 
-def _validate_classes(classes):
-    if not isinstance(classes, list):
-        raise ValidationException("Field `classes` must be a list of objects.")
-    for idx, cls in enumerate(classes):
-        _validate_class(cls, idx)
+def validate_dataset_update(dataset):
+    if not isinstance(dataset, dict):
+        raise ValidationException("Dataset must be a dictionary.")
+    _check_dict_structure(
+        dataset,
+        [
+            ("name", False),
+            ("description", False),
+            ("public", False),
+        ],
+        "dataset dictionary",
+    )
+
+    # Name
+    if "name" in dataset:
+        _validate_dataset_name(dataset["name"])
+
+    # Description (optional)
+    if "description" in dataset and dataset["description"] is not None:
+        _validate_dataset_description(dataset["description"])
+
+    # Publicity
+    if "public" in dataset:
+        _validate_dataset_public(dataset["public"])
 
 
-def _validate_class(cls, idx):
+def validate_class(cls, idx=None, recordings_required=True):
+    """ Validate the contents of a class dictionary
+
+    :param cls: the class to check
+    :param idx: the index of this class if it is part of a list of classes,
+                otherwise None if it's a standalone class
+    :param recordings_required: True if this class must have a `recordings` element
+    :raises: `ValidationException` if the class definition has a problem
```

Comment:
We should use the same docstring formatting *(argument and return values documentation in this case)* throughout the project.